python/hsfs/engine/python.py

```diff
@@ -414,20 +420,33 @@ def profile(
 
         final_stats = []
         for col in stats.keys():
-            stats_dict = stats[col]
-            stat = self._convert_pandas_statistics(stats_dict)
+            stat = self._convert_pandas_statistics(stats[col])
+            stat["isDataTypeInferred"] = "false"
+            stat["column"] = col.split(".")[-1]
+            stat["completeness"] = 1
+
+            # set data type
             if isinstance(df.dtypes[col], type(np.dtype(np.float64))):
                 stat["dataType"] = "Fractional"
+            elif isinstance(df.dtypes[col], type(np.dtype(np.int64))):
+                stat["dataType"] = "Integral"
+            elif isinstance(df.dtypes[col], type(np.dtype(np.bool_))):
+                stat["dataType"] = "Boolean"
             elif isinstance(df.dtypes[col], type(np.dtype(object))):
                 stat["dataType"] = "String"
             else:
-                stat["dataType"] = "Integral"
-            stat["isDataTypeInferred"] = "false"
-            stat["column"] = col.split(".")[-1]
-            stat["completeness"] = 1
+                print(
```

Comment:
If this print is reporting an error to the user we should set the `file=sys.stderr` argument in the print.
Also @manu-sj recently added support for pyarrow types in the insert. It looks like this part of the code also needs some fixing regards to that. Don't need to fix that in this PR but please sync up with him and create a JIRA if needed.